### PR TITLE
BookB-Pset_GeoScienceFeatureTypeRockMass

### PR DESCRIPTION
**Pull request by larswik**
# Pset: Pset_GeoScienceFeatureTypeRockMass

## Description: 

Properties for rock mass.

| Property | Datatype | Description |
|----------|----------|-------------|
|HoekBrownMi|IfcReal|Material constant of intact rock.|
|RQD|IfcPositiveRatioMeasure|Quotient of the cumulative length of cores with a length greater than 10 cm, by the total length of the core pass with a length greater than or equal to 1m.|
|RMR|IfcInteger|The rock mass rating (RMR) is a geomechanical classification system for rocks, developed by Z. T. Bieniawski.|
|QValue|IfcReal|The rock mass quality (Q) is a very sensitive index and its value varies from 0.001 to 1000. Use of the Q-system is specifically recommended for tunnels and caverns with an arched roof.|
|GSI|IfcInteger|The Geological Strength Index (GSI): A characterization tool for assessing engineering properties for rock masses.|
|YoungModulus|According to existing definition|According to existing definition|
|Cohesion|IfcPressureMeasure|Cohesive shear strength of a rock or soil that is independent of interparticle friction.|
|FrictionAngle|IfcPlaneAngleMeasure|Derived from the Mohr-Coulomb failure criterion and used to describe the friction shear resistance of ground materials, together with the normal effective stress.|
|K0|IfcReal|Lateral Earth Pressure Coefficient.|

### CODE REVIEW

File: IFC4x3/Properties/e/Emodulus_10zwR$ze1ElBZzkStdNfro/DocProperty.xml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="Emodulus_10zwRze1ElBZzkStdNfro" Name="Emodulus" UniqueId="40f7a6ff-f680-4ebc-b8fd-b9cde75e9d72" PrimaryDataType="IfcModulusOfElasticityMeasure" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/e/Emodulus_10zwR$ze1ElBZzkStdNfro/DocProperty.xml]: Reuse YoungModulus.

File: IFC4x3/Properties/g/GSI_1mKh3XC4vAkAZD1jzmwtzK/DocProperty.xml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="GSI_1mKh3XC4vAkAZD1jzmwtzK" Name="GSI" UniqueId="7052b0e1-304e-4ab8-a8cd-06df70eb7f54" PropertyType="p_boundedvalue" PrimaryDataType="IfcInteger" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/g/GSI_1mKh3XC4vAkAZD1jzmwtzK/DocProperty.xml]: Bounded value?

File: IFC4x3/Properties/h/HoekBrownMi_1Aq22czzr7AuXPKS91fYGu/DocProperty.xml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="HoekBrownMi_1Aq22czzr7AuXPKS91fYGu" Name="HoekBrownMi" UniqueId="4ad020a6-f7dd-472b-8859-51c241a62438" PropertyType="p_boundedvalue" PrimaryDataType="IfcReal" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/h/HoekBrownMi_1Aq22czzr7AuXPKS91fYGu/DocProperty.xml]: Bounded value?

The description seems a bit short for it. Please review if it fits as is.

File: IFC4x3/Properties/k/K0_3LHWuSc4f4oAmYOduGjUmW/DocProperty.xml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="K0_3LHWuSc4f4oAmYOduGjUmW" Name="K0" UniqueId="d5460e1c-984a-44c8-ac22-627e10b5ec20" PrimaryDataType="IfcReal" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/k/K0_3LHWuSc4f4oAmYOduGjUmW/DocProperty.xml:2]: Is this supposed to be reused? The description does not match the other K0 from here https://github.com/bSI-InfraRoom/IFC-Specification/pull/630.

File: IFC4x3/Properties/q/QValue_1bhEISWbXCwxu8IMpl$sWn/Documentation.md [hunk 5]
```diff
@@ -0,0 +1 @@
+The rock mass quality (Q) is a very sensitive index and its value varies from 0.001 to 1000. Use of the Q-system is specifically recommended for tunnels and caverns with an arched roof.
```
Comment (SergejMuhic) [inline IFC4x3/Properties/q/QValue_1bhEISWbXCwxu8IMpl$sWn/Documentation.md:1]: What is it for?

File: IFC4x3/Properties/r/RMR_2qdGAN$0PBMO5OKXlwlUlX/DocProperty.xml [hunk 6]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="RMR_2qdGAN0PBMO5OKXlwlUlX" Name="RMR" UniqueId="b49d0297-fc06-4b59-8158-521bfabdebe1" PrimaryDataType="IfcInteger" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/r/RMR_2qdGAN$0PBMO5OKXlwlUlX/DocProperty.xml]: I would prefer RockMassRating for the name. Non domain experts can more easily identify what it means and can search for it.

File: IFC4x3/Properties/r/RQD_2o32BT3ZX4bO$w85z$rZ25/DocProperty.xml [hunk 7]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="RQD_2o32BT3ZX4bOw85zrZ25" Name="RQD" UniqueId="b20c22dd-0e38-4495-8ffa-205f7fd63085" PrimaryDataType="IfcPositiveRatioMeasure" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/r/RQD_2o32BT3ZX4bO$w85z$rZ25/DocProperty.xml]: A more descriptive name would be welcome.

File: IFC4x3/Properties/r/RQD_2o32BT3ZX4bO$w85z$rZ25/Documentation.md [hunk 8]
```diff
@@ -0,0 +1 @@
+Quotient of the cumulative length of cores with a length greater than 10 cm, by the total length of the core pass with a length greater than or equal to 1m.
```
Comment (SergejMuhic) [inline IFC4x3/Properties/r/RQD_2o32BT3ZX4bO$w85z$rZ25/Documentation.md]: Not sure what is meant by "cores".

File: IFC4x3/Properties/e/Emodulus_10zwR$ze1ElBZzkStdNfro/DocProperty.xml (hunk 1 above)
Comment (larswik) [reply IFC4x3/Properties/e/Emodulus_10zwR$ze1ElBZzkStdNfro/DocProperty.xml]: Same comment as many other places. Hopefully with one generic answer.

File: IFC4x3/Properties/g/GSI_1mKh3XC4vAkAZD1jzmwtzK/DocProperty.xml (hunk 2 above)
Comment (larswik) [reply IFC4x3/Properties/g/GSI_1mKh3XC4vAkAZD1jzmwtzK/DocProperty.xml]: Double check this! @j-weil.

File: IFC4x3/Properties/h/HoekBrownMi_1Aq22czzr7AuXPKS91fYGu/DocProperty.xml (hunk 3 above)
Comment (larswik) [reply IFC4x3/Properties/h/HoekBrownMi_1Aq22czzr7AuXPKS91fYGu/DocProperty.xml]: Double check.

File: IFC4x3/Properties/k/K0_3LHWuSc4f4oAmYOduGjUmW/DocProperty.xml (hunk 4 above)
Comment (larswik) [reply IFC4x3/Properties/k/K0_3LHWuSc4f4oAmYOduGjUmW/DocProperty.xml:2]: Double check.

File: IFC4x3/Properties/q/QValue_1bhEISWbXCwxu8IMpl$sWn/Documentation.md (hunk 5 above)
Comment (larswik) [reply IFC4x3/Properties/q/QValue_1bhEISWbXCwxu8IMpl$sWn/Documentation.md:1]: Elaborate description?

File: IFC4x3/Properties/r/RMR_2qdGAN$0PBMO5OKXlwlUlX/DocProperty.xml (hunk 6 above)
Comment (larswik) [reply IFC4x3/Properties/r/RMR_2qdGAN$0PBMO5OKXlwlUlX/DocProperty.xml]: Similar as a few other comments as well.

File: IFC4x3/Properties/r/RQD_2o32BT3ZX4bO$w85z$rZ25/DocProperty.xml (hunk 7 above)
Comment (larswik) [reply IFC4x3/Properties/r/RQD_2o32BT3ZX4bO$w85z$rZ25/DocProperty.xml]: Similar as a few other comments as well.

File: IFC4x3/Properties/r/RQD_2o32BT3ZX4bO$w85z$rZ25/Documentation.md (hunk 8 above)
Comment (larswik) [reply IFC4x3/Properties/r/RQD_2o32BT3ZX4bO$w85z$rZ25/Documentation.md]: Maybe an elaboration of the description is needed?

File: IFC4x3/Properties/e/Emodulus_10zwR$ze1ElBZzkStdNfro/DocProperty.xml (hunk 1 above)
Comment (larswik) [reply IFC4x3/Properties/e/Emodulus_10zwR$ze1ElBZzkStdNfro/DocProperty.xml]: Fixed in [e71e7f6](https://github.com/bSI-InfraRoom/IFC-Specification/pull/629/commits/e71e7f690163e6a5a1a79ed488397ea2222c61d8)

File: IFC4x3/Properties/g/GSI_1mKh3XC4vAkAZD1jzmwtzK/DocProperty.xml (hunk 2 above)
Comment (larswik) [reply IFC4x3/Properties/g/GSI_1mKh3XC4vAkAZD1jzmwtzK/DocProperty.xml]: Yes bounded.

File: IFC4x3/Properties/h/HoekBrownMi_1Aq22czzr7AuXPKS91fYGu/DocProperty.xml (hunk 3 above)
Comment (larswik) [reply IFC4x3/Properties/h/HoekBrownMi_1Aq22czzr7AuXPKS91fYGu/DocProperty.xml]: Property removed.

File: IFC4x3/Properties/k/K0_3LHWuSc4f4oAmYOduGjUmW/DocProperty.xml (hunk 4 above)
Comment (larswik) [reply IFC4x3/Properties/k/K0_3LHWuSc4f4oAmYOduGjUmW/DocProperty.xml:2]: Await merge of #630 and reuse that.

File: IFC4x3/Properties/q/QValue_1bhEISWbXCwxu8IMpl$sWn/Documentation.md (hunk 5 above)
Comment (larswik) [reply IFC4x3/Properties/q/QValue_1bhEISWbXCwxu8IMpl$sWn/Documentation.md:1]: Elaborated.

File: IFC4x3/Properties/r/RMR_2qdGAN$0PBMO5OKXlwlUlX/DocProperty.xml (hunk 6 above)
Comment (larswik) [reply IFC4x3/Properties/r/RMR_2qdGAN$0PBMO5OKXlwlUlX/DocProperty.xml]: Renamed.

File: IFC4x3/Properties/r/RQD_2o32BT3ZX4bO$w85z$rZ25/DocProperty.xml (hunk 7 above)
Comment (larswik) [reply IFC4x3/Properties/r/RQD_2o32BT3ZX4bO$w85z$rZ25/DocProperty.xml]: Renamed.

File: IFC4x3/Properties/r/RQD_2o32BT3ZX4bO$w85z$rZ25/Documentation.md (hunk 8 above)
Comment (larswik) [reply IFC4x3/Properties/r/RQD_2o32BT3ZX4bO$w85z$rZ25/Documentation.md]: Renamed + changed "cores" to "drill cores".

**Pull request merged**